src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.Update.cs

```diff
@@ -343,17 +371,21 @@ await RepeatIOAsync(
                         IOUtilities.PerformIO(() => _service._ioService.Delete(new FileInfo(tempFilePath)));
                     }
                 }, cancellationToken).ConfigureAwait(false);
-
-            LogInfo("Writing database file completed");
         }
 
+        private static FileInfo GetBinaryFileInfo(FileInfo databaseFileInfo)
+            => new FileInfo(Path.ChangeExtension(databaseFileInfo.FullName, ".bin"));
+
         private async Task<TimeSpan> PatchLocalDatabaseAsync(FileInfo databaseFileInfo, CancellationToken cancellationToken)
         {
             LogInfo("Patching local database");
 
             LogInfo("Reading in local database");
             // (intentionally not wrapped in IOUtilities.  If this throws we want to restart).
-            var databaseBytes = _service._ioService.ReadAllBytes(databaseFileInfo.FullName);
+            var databaseBinaryFileInfo = GetBinaryFileInfo(databaseFileInfo);
+            var isBinary = _service._ioService.Exists(databaseBinaryFileInfo);
+            var databaseBytes = _service._ioService.ReadAllBytes(isBinary ? databaseBinaryFileInfo.FullName : databaseFileInfo.FullName);
```

Comment:
there is a race-condition here between checking existence and then reading.  could we just attempt to ReadAllBytes of hte binary file, and fall over it we get null there?